### PR TITLE
[2855] hpitt snags

### PR DESCRIPTION
**Pull request by archferns**
### Context
Tweak the HPITT import task to be able to process more/handle some formatting issues.

### Changes proposed in this pull request
1. ~Add synonyms for some Institutions [here](https://github.com/DFE-Digital/register-trainee-teachers/pull/1545/files#diff-b56efc264abfac11597539934764ce3fe02b9ce1141d322eb10c40932eaf8722R12-R16)~ Add a  mapping for some institutions [here](https://github.com/DFE-Digital/register-trainee-teachers/pull/1545/files#diff-7b6263050ddc34b8ed995a32e34ad36618ee26b40601d708a324522d2074aad5R3-R27)
2. Add a formatting/mapping for post codes
3. Add support for international addresses, where provided
4. ensure that completed sections are marked as complete
5. Add course subject mappings [here](https://github.com/DFE-Digital/register-trainee-teachers/pull/1545/files#diff-72e177d5ce0b411028ab0d86c259764d5efb9496ffa712e0e7f517380f4b9970R1-R22)
6. Add disability mapping [here](https://github.com/DFE-Digital/register-trainee-teachers/pull/1545/files#diff-efa674f9ed3cdf8511556161aee3fc45b5a131724b6db4457f49db0e9139d947R1-R17)

### Guidance to review
It would be helpful to validate/sense check the mappings added in.

Also adding a quick and dirty script here to run against the Teach first trainee list and find any validation errors using the all powerful `TrnSubmissionForm`:
```ruby
trainees_with_errors = Provider.find_by(code: TEACH_FIRST_PROVIDER_CODE).trainees.map do |trainee|
  trn_submission_form = TrnSubmissionForm.new(trainee: trainee)
  trn_submission_form.form_validators.map do |validator|
    form = validator.second[:form].constantize.new(trainee)
    form.valid? ? nil : [trainee.slug, form.errors.messages]
  end.reject(&:blank?)
end.reject(&:blank?)
# => [[["zeS2BgPKFg3EbsJ4eyj8PE9E", {:nationality_names=>["Select at least one nationality"]}], ...

```

### CODE REVIEW

**Comment by archferns** [comment]
[HPITT Snags](https://trello.com/c/Usl5EtDK/2855-hpitt-snags)

File: app/lib/hpitt.rb [hunk 1]
```diff
@@ -87,6 +93,40 @@ def build_trainee(csv_row)
         column_mapper[key].call(value)
       end
 
+      if trainee.international_address.present?
```
Comment (darokel) [inline app/lib/hpitt.rb:96]: Feels like we could break some of this up into their own methods to keep `build_trainee` from getting bigger than it already is but this isn't a blocker at all for this PR

File: app/lib/hpitt.rb [hunk 2]
```diff
@@ -176,10 +255,38 @@ def to_nationality_ids(raw_string)
...
     end
+
+    def normalise_string(raw_string)
+      raw_string
+      .downcase
+      .gsub(/\(.*\)/, "")
+      .split
+      .reject { |word| REJECTED_WORD_LIST.include? word }
+      .join(" ")
+      .gsub(/[^\w]/, "")
```
Comment (darokel) [inline app/lib/hpitt.rb:277]: something for the future - we should probably label the regexs so it's easier to mentally parse, we've got quite a few in here

**Review by darokel — APPROVED** [review]
left a couple of comments but not blockers for this PR

**Pull request merged**